plugin crash everytime on tf plan when upgrading to version >= 1.15.0

**Describe the bug**
Terraform plan crashes after upgrade from 1.14.2 to any other higher versions (tested with v.2.0.0 incl.)

**Requirements for and issue**
code snippet:
![image](https://github.com/jfrog/terraform-provider-xray/assets/15525555/b212340b-fa5c-4250-94a2-86a0d2270afd)

- JFrog SaaS instance (EnterpriseX, without Advanced DevSecOps)
- artifactory version: 7.70.2
- xray version: 3.82.10
- tf version: v1.5.7

**Expected behavior**
Works still after the upgrade :)

**Additional context**
![image](https://github.com/jfrog/terraform-provider-xray/assets/15525555/13763b01-172e-4bb9-966c-b97633aad7ca)

Hello,

I also have problems with Xray provider upgrade from version to 1.12 to any version from 1.15 to 2.0.0.

In the plan I have a lot of UpgradeResourceState errors like this:

```
│ Error: Request cancelled
│ 
│   with module.watch-critical-all-repos-lyft.xray_security_policy.critical_cvss,
│   on .terraform/modules/watch-critical-all-repos-lyft/ops/terraform/modules/xray-security/watch-critical/main.tf line 10, in resource "xray_security_policy" "critical_cvss":
│   10: resource "xray_security_policy" "critical_cvss" {
│ 
│ The plugin.(*GRPCProvider).UpgradeResourceState request was cancelled.
```

And as a result:

```
Stack trace from the terraform-provider-xray_v2.0.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb03bcb]

goroutine 81 [running]:
github.com/jfrog/terraform-provider-xray/pkg/xray.resourceXrayRepositoryConfig.func7(0x0, {0xc000138778, 0x4})
        github.com/jfrog/terraform-provider-xray/pkg/xray/resource_xray_repository_config.go:354 +0x10b
github.com/jfrog/terraform-provider-xray/pkg/xray.resourceXrayRepositoryConfig.func8(0xc0007864e0, {0xc000451c70, 0x6}, {0xc000138778, 0x4})
        github.com/jfrog/terraform-provider-xray/pkg/xray/resource_xray_repository_config.go:381 +0x2b8
github.com/jfrog/terraform-provider-xray/pkg/xray.resourceXrayRepositoryConfig.func12({0x1c?, 0xcaae12?}, {{0xc0007b3660?, 0x203000?}, 0xc0007864e0?, 0x0?}, 0xc0003900c0?, {0xc000451c70, 0x6}, {0xc000138778, ...})
        github.com/jfrog/terraform-provider-xray/pkg/xray/resource_xray_repository_config.go:430 +0x117
github.com/jfrog/terraform-provider-xray/pkg/xray.resourceXrayRepositoryConfig.func14({0xddb5e8, 0xc000060b40}, 0xc00030e480, {0xc0b420?, 0xc000487740})
        github.com/jfrog/terraform-provider-xray/pkg/xray/resource_xray_repository_config.go:483 +0x64a
github.com/jfrog/terraform-provider-shared/util/sdk.applyTelemetry.func1({0xddb5e8?, 0xc000060b40}, 0x0?, {0xc0b420?, 0xc000487740})
        github.com/jfrog/terraform-provider-shared@v1.18.0/util/sdk/sdk.go:234 +0x263
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc00049ec40, {0xddb620, 0xc000518180}, 0xd?, {0xc0b420, 0xc000487740})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:750 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00049ec40, {0xddb620, 0xc000518180}, 0xc0000b7040, {0xc0b420, 0xc000487740})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:1044 +0x59e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00030da58, {0xddb620?, 0xc000518030?}, 0xc0003907c0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/grpc_provider.go:616 +0x497
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0004b63c0, {0xddb620?, 0xc0002cea80?}, 0xc000176660)
        github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/tf5server/server.go:751 +0x49e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xc60400?, 0xc0004b63c0}, {0xddb620, 0xc0002cea80}, 0xc000368070, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:386 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00047c000, {0xdde2a0, 0xc0001024e0}, 0xc000355b00, 0xc000487560, 0x128ef90, 0x0)
        google.golang.org/grpc@v1.56.1/server.go:1337 +0xde3
google.golang.org/grpc.(*Server).handleStream(0xc00047c000, {0xdde2a0, 0xc0001024e0}, 0xc000355b00, 0x0)
        google.golang.org/grpc@v1.56.1/server.go:1714 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.56.1/server.go:959 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.56.1/server.go:957 +0x18c

Error: The terraform-provider-xray_v2.0.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
``` 

And x-ray config of repository is like this:

```
terraform {
  required_providers {
    artifactory = {
      source  = "registry.terraform.io/jfrog/artifactory"
      version = ">= 6.10.1, < 9.0.0"
    }
    xray = {
      source  = "registry.terraform.io/jfrog/xray"
      version = ">= 1.11.1, <= 2.0.0"
    }
  }
}

resource "artifactory_local_maven_repository" "repo" {
  key                  = var.key
  repo_layout_ref      = var.layout_ref
  description          = "**MANAGED BY TERRAFORM** ${var.description}"
  notes                = "Owner: ${var.owner}"
  includes_pattern     = var.includes_pattern
  excludes_pattern     = var.excludes_pattern
  priority_resolution  = var.priority_resolution
  handle_releases      = var.handle_releases
  handle_snapshots     = var.handle_snapshots
  property_sets        = ["artifactory"]
  download_direct      = var.download_direct
  xray_index           = var.xray_index
  blacked_out          = var.blacked_out
  checksum_policy_type = var.checksum_policy_type
}

resource "xray_repository_config" "local-maven-xray-config" {
  provider  = xray
  repo_name = var.key
  config {
    retention_in_days   = 90
  }
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,4 +1,4 @@
-## 2.0.2 (November 1, 2023)
+## 2.0.2 (November 1, 2023). Tested on Artifactory 7.71.3 and Xray 3.83.10
 
 BUG FIXES:
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,12 @@
+## 2.0.2 (November 1, 2023)
+
+BUG FIXES:
+
+* resource/xray_repository_config: fix provider crash after upgrading from 1.12.0 to >=1.15.0.
+
+PR: [#145](https://github.com/jfrog/terraform-provider-xray/pull/145)
+Issue: [#141](https://github.com/jfrog/terraform-provider-xray/issues/141) and [#144](https://github.com/jfrog/terraform-provider-xray/issues/144)
+
 ## 2.0.1 (October 12, 2023). Tested on Artifactory 7.71.3 and Xray 3.83.10
 
 SECURITY:

--- a/go.mod
+++ b/go.mod
@@ -10,7 +10,7 @@ require (
 	github.com/hashicorp/terraform-plugin-log v0.9.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.27.0
 	github.com/hashicorp/terraform-plugin-testing v1.4.0
-	github.com/jfrog/terraform-provider-shared v1.18.0
+	github.com/jfrog/terraform-provider-shared v1.20.2
 	golang.org/x/exp v0.0.0-20230522175609-2e198f4a06a1
 	golang.org/x/text v0.13.0
 )

--- a/pkg/xray/resource_xray_repository_config.go
+++ b/pkg/xray/resource_xray_repository_config.go
@@ -377,7 +377,7 @@ func resourceXrayRepositoryConfig() *schema.Resource {
 			m["vuln_contextual_analysis"] = *repoConfig.VulnContextualAnalysis
 		}
 
-		if slices.Contains(exposuresPackageTypes(xrayVersion), packageType) {
+		if repoConfig.Exposures != nil && slices.Contains(exposuresPackageTypes(xrayVersion), packageType) {
 			m["exposures"] = packExposures(repoConfig.Exposures, packageType)
 		}
 

